packages/sharing-smart-contract/test/subscription.test.js

```diff
@@ -23,6 +25,103 @@ describe('Subscription.sol', () => {
+
+      expect(ethers.toNumber(subscriptionInfo)).to.equal(expectedEndDate);
+      // there is no transaction fees on bellecour
+      expect(subscriberBalanceAfter).to.equal(subscriberBalanceBefore - subscriptionPrice);
+    });
+
+    it('should revert if subscription parameters are not set', async () => {
+      const { subscriptionContract, addr1 } = await loadFixture(deploySCFixture);
+      const tx = await subscriptionContract.connect(addr1).createCollection();
+      const receipt = await tx.wait();
+      const tokenId = ethers.toNumber(receipt.logs[0].args[2]);
+
+      await expect(
+        subscriptionContract
+          .connect(addr1)
+          .subscribeTo(tokenId, { value: ethers.parseEther('0.1') }),
+      ).to.be.revertedWith('Subscription parameters not set');
+    });
+
+    it('should revert if the subscription price is lower than required', async () => {
+      const { subscriptionContract, addr1 } = await loadFixture(deploySCFixture);
+      const tx = await subscriptionContract.connect(addr1).createCollection();
+      const receipt = await tx.wait();
+      const tokenId = ethers.toNumber(receipt.logs[0].args[2]);
+      const subscriptionParams = {
+        price: ethers.parseEther('1'),
+        duration: 30,
+      };
+      await subscriptionContract.connect(addr1).setSubscriptionParams(tokenId, subscriptionParams);
+
+      await expect(
+        subscriptionContract
+          .connect(addr1)
+          .subscribeTo(tokenId, { value: ethers.parseEther('0.5') }),
+      ).to.be.revertedWith('Wrong amount sent');
+    });
+
+    it('should revert if the subscription price is not equal to value sent', async () => {
+      const { subscriptionContract, addr1 } = await loadFixture(deploySCFixture);
+      const tx = await subscriptionContract.connect(addr1).createCollection();
+      const receipt = await tx.wait();
+      const tokenId = ethers.toNumber(receipt.logs[0].args[2]);
+
+      const subscriptionPrice = ethers.parseEther('0.5');
+      const tokenSended = ethers.parseEther('1.3'); // Send more money than required
+      const subscriptionParams = {
+        price: subscriptionPrice,
+        duration: 15,
+      };
+      const SubscriptionOptionsTx = await subscriptionContract
+        .connect(addr1)
+        .setSubscriptionParams(tokenId, subscriptionParams);
+      await SubscriptionOptionsTx.wait();
+
+      const subscriberBalanceBefore = await ethers.provider.getBalance(addr1.address);
+      const subscriptionTx = subscriptionContract
+        .connect(addr1)
+        .subscribeTo(tokenId, { value: tokenSended });
+      await expect(subscriptionTx).to.be.revertedWith('Wrong amount sent');
+      const subscriberBalanceAfter = await ethers.provider.getBalance(addr1.address);
+
+      // there is no transaction fees on bellecour
+      expect(subscriberBalanceAfter).to.equal(subscriberBalanceBefore);
+    });
+  });
+
```

Comment:
you could add a test case to verify that an attempt to subscribe with insufficient funds reverts with an appropriate error message.

Reply:
Yeah but it's the validator that verify that i have enough funds not the ContentCreator Contract